### DISCUSSION TITLE
NFR: ability to do some measurements

### DISCUSSION THREAD

**Discussion by cyberhck** [99designs/gqlgen#1933] [Q&A]
First of all, gqlgen is THE best gql client as of now period. It's mainly because of TypeSafety and schema first design, I love it. There's no question to it. (subscription doesn't seem to be have a documentation though)

### What happened?
I want to monitor my queries. For each queries (resolvers called), I'd like to get the time taken before and after for each of those, (I don't want to put this call on each resolvers however), that way I could simply do some measurements and make sure my queries and mutations are behaving correctly.

### What did you expect?
I was under the impression that this was going to be a piece of :cake: , but turns out it's kinda tricky.

I know with the typesafety stuff, it might be a tad difficult, but what do you recommend? Is there something we can do?

**Comment by philipjscott** [comment]
I believe GQL already provides a way to measure performance: https://github.com/99designs/gqlgen/search?q=Tracer

See:
- https://github.com/99designs/gqlgen/blob/3a31a752df764738b1f6e99408df3b169d514784/graphql/handler/debug/tracer.go
- https://github.com/99designs/gqlgen/blob/3a31a752df764738b1f6e99408df3b169d514784/example/federation/reviews/server.go

**Comment by cyberhck** [comment]
I'll look into this tomorrow and verify if that lets me do all the measurements :) thanks.

**Comment by cyberhck** [comment]
I think I might have misunderstood, but doesn't https://github.com/99designs/gqlgen/blob/3a31a752df764738b1f6e99408df3b169d514784/graphql/handler/debug/tracer.go#L61
actually allow me to measure performance of the whole query? what about the sub fields? if they have their own resolvers? The above line helps me to at least keep track of what fields are being used, but how about when user is doing 2 query in single request? In that case, it looks like it's not going to be possible, or if there's a nested graph, say a `City` belongs to a `State` in that case, it'll give you result of whole thing, not individual resolvers.
